pkg/restapi/operation/operations_test.go

```diff
@@ -961,69 +980,75 @@ func TestReadDocument(t *testing.T) {
 		require.NoError(t, err)
 
 		urlVars := make(map[string]string)
-		urlVars[vaultIDPathVariable] = testVaultID
+		urlVars[vaultIDPathVariable] = testReferenceID
 		urlVars[docIDPathVariable] = testDocID
 
 		req = mux.SetURLVars(req, urlVars)
 
 		op.readDocumentHandler(failingResponseWriter{}, req)
 
 		require.Contains(t, mockLoggerProvider.MockLogger.AllLogContents,
-			fmt.Sprintf(messages.ReadDocumentFailure, testDocID, testVaultID, messages.ErrVaultNotFound))
+			fmt.Sprintf(messages.ReadDocumentFailure, testDocID, testReferenceID, messages.ErrVaultNotFound))
 	})
 	t.Run("Response writer fails while writing retrieved document", func(t *testing.T) {
 		op := New(memedvprovider.NewProvider())
 
-		createDataVaultExpectSuccess(t, op)
+		vaultID := createDataVaultExpectSuccess(t, op)
 
-		storeEncryptedDocumentExpectSuccess(t, op, testDocID, testEncryptedDocument)
+		storeEncryptedDocumentExpectSuccess(t, op, testDocID, testEncryptedDocument, vaultID)
 
 		request := http.Request{}
 
 		op.readDocumentHandler(failingResponseWriter{},
-			request.WithContext(mockContext{valueToReturnWhenValueMethodCalled: getMapWithValidVaultIDAndDocID()}))
+			request.WithContext(mockContext{valueToReturnWhenValueMethodCalled: getMapWithValidVaultIDAndDocID(vaultID)}))
 
 		require.Contains(t, mockLoggerProvider.MockLogger.AllLogContents,
 			fmt.Sprintf(messages.ReadDocumentSuccess+messages.FailWriteResponse,
-				testDocID, testVaultID, errFailingResponseWriter))
+				testDocID, vaultID, errFailingResponseWriter))
 	})
 }
 
 func Test_writeReadAllDocumentsSuccess(t *testing.T) {
 	t.Run("Fail to marshal all documents", func(t *testing.T) {
 		rr := httptest.NewRecorder()
 
-		writeReadAllDocumentsSuccess(rr, []json.RawMessage{[]byte("NotValid")}, testVaultID)
+		writeReadAllDocumentsSuccess(rr, []json.RawMessage{[]byte("NotValid")}, testReferenceID)
 
 		require.Equal(t, http.StatusInternalServerError, rr.Code)
-		require.Equal(t, fmt.Sprintf(messages.FailToMarshalAllDocuments, testVaultID, "json: error calling "+
+		require.Equal(t, fmt.Sprintf(messages.FailToMarshalAllDocuments, testReferenceID, "json: error calling "+
 			"MarshalJSON for type json.RawMessage: invalid character 'N' looking for beginning of value"),
 			rr.Body.String())
 	})
 }
 
-func createDataVaultExpectSuccess(t *testing.T, op *Operation) {
+// return created test vault ID
```

Comment:
Should be `returns created test vault ID`

Reply:
fixed